ivy/functional/backends/jax/linear_algebra.py

```diff
@@ -375,19 +373,24 @@ def svdvals(x: JaxArray, /, *, out: Optional[JaxArray] = None) -> JaxArray:
     return jnp.linalg.svd(x, compute_uv=False)
 
 
-@with_unsupported_dtypes({"0.4.16 and below": ("complex",)}, backend_version)
+@with_supported_dtypes({"0.4.16 and below": ("float32", "float64")}, backend_version)
 def tensordot(
     x1: JaxArray,
     x2: JaxArray,
     /,
     *,
     axes: Union[int, Tuple[Sequence[int], Sequence[int]]] = 2,
+    batched_modes: Optional[Union[int, Tuple[List[int], List[int]]]] = None,
     out: Optional[JaxArray] = None,
 ) -> JaxArray:
-    x1, x2 = promote_types_of_inputs(x1, x2)
```

Comment:
We should use `ivy.promote_types` to determine the output datatype and cast the result to that dtype. This is necessary so that all the backends return a consistent dtype for a given input.
```python
ret_dtype = ivy.promote_types(x1.dtype, x2.dtype)
```